.palantir/revapi.yml

```diff
@@ -1178,6 +1178,9 @@ acceptedBreaks:
       new: "class org.apache.iceberg.Metrics"
       justification: "Java serialization across versions is not guaranteed"
     org.apache.iceberg:iceberg-core:
+    - code: "java.class.removed"
+      old: "class org.apache.iceberg.MetadataUpdate.RemoveSnapshot"
+      justification: "Removing deprecations for 1.10.0"
```

Comment:
do we have other deprecations that we wanted to remove in 1.10.0 as we typically try to remove such deprecated code in a single PR if possible

Reply:
Some deprecation still have dependencies, but I will try to add more.

Reply:
for example, we need to get https://github.com/apache/iceberg/pull/12692 in to remove deprecated rewriter API